### PR TITLE
Added some as-yet-unimplemented methods to the base compute driver.

### PR DESCRIPTION
**Pull request by alex**
This is so drivers can have a common design to implement against.

### CODE REVIEW

File: libcloud/compute/base.py [hunk 1]
```diff
@@ -780,6 +808,33 @@ def list_volumes(self):
...
+    def snapshot_volume(self, volume, name):
+        """
+        Creates a snapshot of the storage volume.
+
+        @rtype: L{VolumeSnapshot}
+        """
+        raise NotImplementedError(
+            'snapshot_volume not implemented for this driver')
+
+    def destroy_snapshot(self, snapshot):
```
Comment (Kami) [inline libcloud/compute/base.py]: This one should probably be called something like `destroy_volume_snapshot` so users knows it's related to volume and not virtual machine snapshots.

File: libcloud/compute/base.py [hunk 2]
```diff
@@ -780,6 +808,33 @@ def list_volumes(self):
...
+    def list_volume_snapshots(self, volume):
+        """
+        List snapshots for a storage volume.
+
+        @rtype: C{list} of L{VolumeSnapshot}
+        """
+        raise NotImplementedError(
+            'list_volume_snapshots not implemented for this driver')
+
+    def snapshot_volume(self, volume, name):
```
Comment (Kami) [inline libcloud/compute/base.py]: Maybe `create_volume_snapshot` to be consistent with other method names?

I'm not too opinionated about it though.

File: libcloud/compute/base.py (hunk 1 above)
Comment (Kami) [inline libcloud/compute/base.py]: Besides that, LGTM

**Comment by alex** [comment]
Updated to resolve the issues.

**Comment by Kami** [comment]
Merged, thanks.

**Pull request closed without merging**